app/controllers/admin/announcements_controller.rb

```diff
@@ -32,8 +32,12 @@ def update
   end
 
   def destroy
-    redirect_to admin_announcements_path,
-                success: t('.success', title: @announcement.title) if @announcement.destroy
+    if @announcement.destroy
+      redirect_to admin_announcements_path,
+                  success: t('.success', title: @announcement.title)
+    else
+      redirect_to admin_announcements_path, danger: t('.failure')
```

Comment:
Could I replace [this method we use to show flash messages](https://github.com/Coursemology/coursemology2/blob/master/app/themes/default/views/layouts/default.html.slim#L58) with a helper? This is so we can render a partial or something to display errors with markup, instead of repeating this everywhere:

```ruby
danger: t('.failure', errors: @announcement.errors.full_messages.join(', '))
```